packages/lesswrong/components/ultraFeed/UltraFeedSettings.tsx

```diff
@@ -121,6 +133,143 @@ const styles = defineStyles('UltraFeedSettings', (theme: ThemeType) => ({
   },
 }));
 
+// Helper components for Simple and Advanced views
+interface SimpleViewProps {
+  exploreExploitBiasProps: any;
+  sourceWeights: SettingsFormState['sourceWeights'];
+  sourceWeightErrors: Record<FeedItemSourceType, string | undefined>;
+  onSourceWeightChange: (key: FeedItemSourceType, value: number | string) => void;
+  truncationGridProps: any;
+  miscSettingsProps: any;
+}
+
+const SimpleView: React.FC<SimpleViewProps> = ({
+  exploreExploitBiasProps,
+  sourceWeights,
+  sourceWeightErrors,
+  onSourceWeightChange,
+  truncationGridProps,
+  miscSettingsProps,
+}) => (
+  <>
+    <ExploreExploitBiasSettings {...exploreExploitBiasProps} />
+    <SourceWeightsSettings
+      weights={sourceWeights}
+      errors={sourceWeightErrors}
+      onChange={onSourceWeightChange}
+    />
+    <TruncationGridSettings {...truncationGridProps} defaultOpen={true} />
+    <MiscSettings {...miscSettingsProps} defaultOpen={false} />
+  </>
+);
+
+interface AdvancedViewProps {
+  sourceWeights: SettingsFormState['sourceWeights'];
+  sourceWeightErrors: Record<FeedItemSourceType, string | undefined>;
+  onSourceWeightChange: (key: FeedItemSourceType, value: number | string) => void;
+  advancedTruncationProps: any;
+  commentScoringFormValues: CommentScoringFormState;
+  commentScoringErrors: any; // Consider defining a more specific type
```

Comment:
I like the comment here

Reply:
(note that I haven't looked into the situation with the Zod error types, though I'm a bit worried about whatever's going on there)